lib/src/modal/modal.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, Input, ViewEncapsulation, ComponentRef, HostListener, ElementRef, ViewChild } from "@angular/core";
+import { SebModalBackdropComponent } from "./modal.backdrop";
+import { ModalService } from "./modal.service";
+import { ModalSizeType, ModalPositionType } from "./modal.type";
+import { AnimationEvent } from "@angular/animations";
+import { fadeInAnimation } from "./animation";
+
+@Component({
+    selector: "sebng-modal",
+    styleUrls: ["./modal.component.scss"],
+    templateUrl: "./modal.component.html",
+    encapsulation: ViewEncapsulation.None,
+    animations: [fadeInAnimation],
+})
+export class ModalComponent {
+    @Input() id?: string;
+    @Input() size?: ModalSizeType;
+    @Input() center?: boolean;
+    @Input() position: ModalPositionType;
+    @Input() fullscreen?: boolean;
+    @Input() backdropDismiss?: boolean = true;
+    @Input() escapeKeyDismiss?: boolean = true;
+    @Input() className?: string;
+    @Input() ariaLabel?: string;
+    @Input() ariaDescribedby?: string;
+    @Input() animationDuration?: string = ".15s";
+    @ViewChild("modalRef") modalRef: ElementRef;
+    backDropRef: ComponentRef<SebModalBackdropComponent>;
+    toggle: boolean; // toggle is required to enable the open or close animation
+
+    constructor(private modalService: ModalService) {}
+
+    /**
+     * construct the class names that needs to be appended to the modal depending on the inputs requested
+     * @returns { [key: string]: boolean } key value pair for the positions of the modal
+     */
+    get modalPosition(): { [key: string]: boolean } {
```

Comment:
You can use NGClass type from angular as the return type here since you're using it for ngClass